src/Schema/OracleSchemaManager.php

```diff
@@ -320,6 +320,109 @@ private function getQuotedIdentifierName($identifier)
         return $identifier;
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function listTables()
+    {
+        $currentDatabase = $this->_conn->getDatabase();
+
+        if ($currentDatabase === null) {
+            throw new Exception('Invalid database name.');
+        }
+
+        $tableNames = $this->listTableNames();
```

Comment:
Why do we need to retrieve table names separately? Couldn't they be derived from the list of column names?

Reply:
Done